comictaggerlib/settingswindow.py

```diff
@@ -235,6 +234,100 @@ def disconnect_signals(self) -> None:
         self.twLiteralReplacements.cellChanged.disconnect()
         self.twValueReplacements.cellChanged.disconnect()
 
+        self.sources: dict = {}
+        self.generate_source_option_tabs()
+
+    def generate_source_option_tabs(self) -> None:
+        def format_internal_name(int_name: str = "") -> str:
```

Comment:
I want a test written for this so I can see exactly the end result should look like